swan-lake/learn/observing-ballerina-code.md

```diff
@@ -65,70 +67,61 @@ service /hello on new http:Listener(9090) {
-
-```bash
-$ bal run hello_world_service.bal --b7a.observability.enabled=true
-
-[ballerina/http] started HTTP/WS listener 0.0.0.0:9797
-ballerina: started Prometheus HTTP listener 0.0.0.0:9797
-ballerina: started publishing tracers to Jaeger on localhost:5775
-[ballerina/http] started HTTP/WS listener 0.0.0.0:9090
+```toml
+[build-options]
+observabilityIncluded=true
 ```
 
-Redirect the standard output to a file if you want to monitor logs.
+To include the Prometheus and Jaeger extensions into the executable,
+`ballerinax/prometheus` and `ballerinax/jaeger` modules need to be imported in your Ballerina code.
 
-For example:
-
-```bash
-$ nohup ballerina run hello_world_service.bal --b7a.observability.enabled=true > ballerina.log &
+```ballerina
+import ballerinax/prometheus as _;
+import ballerinax/jaeger as _;
 ```
 
-#### Starting the Service Using a Configuration File
-
-Observability of Ballerina service can also be enabled from the configuration. Create a configuration file such as `ballerina.conf` and add the configuration below that starts metrics monitoring and distributed tracing with default 
-settings.
+Observability is disabled by default at runtime as well, and it can be enabled by adding
+the following runtime configurations to the `Config.toml` file.
 
 ```toml
-[b7a.observability.metrics]
-# Flag to enable Metrics
+[ballerina.observe]
 enabled=true
+```
 
-[b7a.observability.tracing]
-# Flag to enable Tracing
-enabled=true
+Alternatively, you can enable metrics and tracing selectively using the following configurations as well.
+
+```toml
+[ballerina.observe]
+metricsEnabled=true
+tracingEnabled=true
 ```
 
-The created configuration file can be passed to the Ballerina program with `--b7a.config.file` option along with
+The created configuration file can be passed to the Ballerina program with `BALCONFIGFILE` environment variable along with
 the path of the configuration file.
 
 ```bash
-$ bal run hello_world_service.bal --b7a.config.file=<path-to-conf>/ballerina.conf
+$ BALCONFIGFILE=<path-to-conf>/Config.toml bal run --observability-included hello_world_service.bal
 
 [ballerina/http] started HTTP/WS listener 0.0.0.0:9797
 ballerina: started Prometheus HTTP listener 0.0.0.0:9797
-ballerina: started publishing tracers to Jaeger on localhost:5775
+ballerina: started publishing traces to Jaeger on localhost:6831
 [ballerina/http] started HTTP/WS listener 0.0.0.0:9090
 ```
 
-Redirect the standard output to a file if you want to monitor logs.
+By default, when Ballerina observability is enabled, Ballerina runtime exposes internal metrics via an HTTP endpoint for
+metrics monitoring and traces will be published to Jaeger. Prometheus should be configured to scrape metrics from
+the metrics HTTP endpoint in Ballerina.
+
+Ballerina logs are logged on to the console. Therefore, the logs need to be redirected to a file, which can then be
```

Comment:
```suggestion
Ballerina logs are logged on the console. Therefore, the logs need to be redirected to a file, which can then be
```

Reply:
Fixed in 146f6434f70aee59d33f47a274e40ed2ac703d6a